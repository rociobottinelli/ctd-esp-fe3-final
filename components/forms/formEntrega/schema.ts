import * as yup from "yup";

export const errorMsg = {
  form: {
    required: "Complete todos los campos.",
  },
  address: {
    required: "La dirección es requerida.",
  },
  state: {
    required: "La provincia es requerida.",
  },
  zipCode: {
    required: "El código postal es requerido.",
  },
};

const entregaSchema = yup
  .object({
    address1: yup.string().required(errorMsg.address.required),
    address2: yup.string().nullable(),
    city: yup.string().required(errorMsg.state.required),
    state: yup.string().required(errorMsg.state.required),
    zipCode: yup.string().required(errorMsg.zipCode.required),
  })
  .required(errorMsg.form.required);

export default entregaSchema;
