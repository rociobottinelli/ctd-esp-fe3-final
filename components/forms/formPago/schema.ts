import * as yup from "yup";
export const errorMsg = {
  form: {
    required: "Complete todos los campos.",
  },
  number: {
    required: "El número es requerido.",
    min: "Por favor ingrese 16 números.",
  },

  nameOnCard: {
    required: "El nombre es requerido.",
  },
  expDate: {
    required: "La fecha de vencimiento es requerida.",
    min: "Por favor ingrese 4 números.",
  },
  cvc: {
    required: "El código CVC es requerido.",
    min: "Por favor ingrese 3 números.",
  },
};

const pagoSchema = yup
  .object({
    number: yup
      .string()
      .min(16, errorMsg.number.min)
      .required(errorMsg.number.required),
    nameOnCard: yup.string().required(errorMsg.nameOnCard.required),
    expDate: yup
      .string()
      .min(4, errorMsg.expDate.min)
      .required(errorMsg.expDate.required),
    cvc: yup
      .string()
      .min(3, errorMsg.cvc.min)
      .required(errorMsg.cvc.required),
  })
  .required(errorMsg.form.required);

export default pagoSchema;
