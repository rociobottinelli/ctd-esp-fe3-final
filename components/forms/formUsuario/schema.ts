import * as yup from "yup";
export const errorMsg = {
  firstname: {
    required: "El nombre es un campo requerido.",
    min: "Por favor ingrese mínimo 2 caracteres.",
    max: "Por favor ingrese mínimo 50 caracteres.",
  },
  lastname: {
    required: "El apellido es un campo requerido.",
    min: "Por favor ingrese mínimo 2 caracteres.",
    max: "Por favor ingrese mínimo 50 caracteres.",
  },
  email: {
    required: "El email es un campo requerido.",
    format: "Por favor ingrese un mail válido.",
  },
};

export const usuarioSchema = yup
  .object({
    name: yup
      .string()
      .required(errorMsg.firstname.required)
      .min(2, errorMsg.firstname.min)
      .max(50, errorMsg.firstname.max),

    lastname: yup
      .string()
      .required(errorMsg.lastname.required)
      .min(2, errorMsg.lastname.min)
      .max(50, errorMsg.lastname.max),

    email: yup
      .string()
      .required(errorMsg.email.required)
      .email(errorMsg.email.format),
  })
  .required();
