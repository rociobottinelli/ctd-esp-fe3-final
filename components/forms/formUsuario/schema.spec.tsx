import { errorMsg, usuarioSchema } from "./schema";


describe("usuarioSchema", () => {
  test("valid input should return no errors", async () => {
    const input = {
      name: "John",
      lastname: "Doe",
      email: "johndoe@example.com",
    };

    await expect(usuarioSchema.validate(input)).resolves.toBe(input);
  });

  test("missing required field should return error", async () => {
    const input = {
      name: "John",
      lastname: "Doe",
      // email is missing
    };

    await expect(usuarioSchema.validate(input)).rejects.toThrow(errorMsg.email.required);
  });

  test("invalid email format should return error", async () => {
    const input = {
      name: "John",
      lastname: "Doe",
      email: "invalid-email", // invalid email format
    };

    await expect(usuarioSchema.validate(input)).rejects.toThrow(errorMsg.email.format);
  });

  test("name below minimum length should return error", async () => {
    const input = {
      name: "J", // below minimum length of 2
      lastname: "Doe",
      email: "johndoe@example.com",
    };

    await expect(usuarioSchema.validate(input)).rejects.toThrow(errorMsg.firstname.min);
  });

  test("name above maximum length should return error", async () => {
    const input = {
      name: "a".repeat(51), // above maximum length of 50
      lastname: "Doe",
      email: "johndoe@example.com",
    };

    await expect(usuarioSchema.validate(input)).rejects.toThrow(errorMsg.firstname.max);
  });

  describe('usuarioSchema', () => {
    it('should validate a valid user object', async () => {
      const user = {
        name: 'John',
        lastname: 'Doe',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).resolves.toEqual(user);
    });
  
    it('should fail validation when name is missing', async () => {
      const user = {
        lastname: 'Doe',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when name is too short', async () => {
      const user = {
        name: 'J',
        lastname: 'Doe',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when name is too long', async () => {
      const user = {
        name: 'John'.repeat(20),
        lastname: 'Doe',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when lastname is missing', async () => {
      const user = {
        name: 'John',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when lastname is too short', async () => {
      const user = {
        name: 'John',
        lastname: 'D',
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when lastname is too long', async () => {
      const user = {
        name: 'John',
        lastname: 'Doe'.repeat(20),
        email: 'johndoe@example.com'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when email is missing', async () => {
      const user = {
        name: 'John',
        lastname: 'Doe'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  
    it('should fail validation when email is not a valid format', async () => {
      const user = {
        name: 'John',
        lastname: 'Doe',
        email: 'invalid-email-format'
      };
      await expect(usuarioSchema.validate(user)).rejects.toThrowErrorMatchingSnapshot();
    });
  });
  
});
