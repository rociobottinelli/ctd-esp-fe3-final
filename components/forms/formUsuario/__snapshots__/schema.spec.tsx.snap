// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usuarioSchema usuarioSchema should fail validation when email is missing 1`] = `"El email es un campo requerido."`;

exports[`usuarioSchema usuarioSchema should fail validation when email is not a valid format 1`] = `"Por favor ingrese un mail válido."`;

exports[`usuarioSchema usuarioSchema should fail validation when lastname is missing 1`] = `"El apellido es un campo requerido."`;

exports[`usuarioSchema usuarioSchema should fail validation when lastname is too long 1`] = `"Por favor ingrese mínimo 50 caracteres."`;

exports[`usuarioSchema usuarioSchema should fail validation when lastname is too short 1`] = `"Por favor ingrese mínimo 2 caracteres."`;

exports[`usuarioSchema usuarioSchema should fail validation when name is missing 1`] = `"El nombre es un campo requerido."`;

exports[`usuarioSchema usuarioSchema should fail validation when name is too long 1`] = `"Por favor ingrese mínimo 50 caracteres."`;

exports[`usuarioSchema usuarioSchema should fail validation when name is too short 1`] = `"Por favor ingrese mínimo 2 caracteres."`;
